opal/mca/memory/patcher/configure.m4

```diff
@@ -45,7 +45,7 @@ AC_DEFUN([MCA_opal_memory_patcher_CONFIG],[
     memory_patcher_have___syscall_prototype=0
 
     AC_MSG_CHECKING([for __curbrk symbol])
-    AC_LINK_IFELSE([AC_LANG_PROGRAM([extern char *__curbrk;],[char *tmp = __curbrk;])],
+    AC_LINK_IFELSE([AC_LANG_PROGRAM([extern char *__curbrk;],[if (*__curbrk) return 1;])],
```

Comment:
Could that cause a segv?  I.e., if `__curbrk` is NULL (or otherwise invalid), won't the dereference cause a problem?  Or is it guaranteed that dereferencing `__curbrk` is always ok?